sub-packages/bionemo-llm/src/bionemo/llm/lightning.py

```diff
@@ -275,11 +278,17 @@ def forward_step(self, batch) -> Tensor:
 
     def training_step(self, batch, batch_idx: Optional[int] = None) -> Tensor:
         """In mcore the loss-function is part of the forward-pass when labels are provided."""
-        return self.forward_step(batch)
+        outputs = self.forward_step(batch)
+        self.train_ppl(outputs["token_logits"], batch["labels"])
+        self.log("train_ppl_step", self.train_ppl)
+        return outputs
 
     def validation_step(self, batch, batch_idx: Optional[int] = None) -> Tensor:
         """In mcore the loss-function is part of the forward-pass when labels are provided."""
-        return self.forward_step(batch)
+        outputs = self.forward_step(batch)
+        self.valid_ppl(outputs["token_logits"], batch["labels"])
+        self.log("valid_ppl_step", self.valid_ppl)
+        return outputs
```

Comment:
Where does `validation_step` get called and how does this interact with megatron parallelism? Just wondering whether torchmetrics would correctly detect the parallelism settings and collect data from all ranks with this configuration

Reply:
Maybe this would be a good place for a coment about `token_logits` being `[s,b,*]` while other things are `[b,s,*]`